datafusion/core/src/physical_optimizer/join_selection.rs

```diff
@@ -157,7 +157,9 @@ fn swap_join_projection(
 }
 
 /// This function swaps the inputs of the given join operator.
-fn swap_hash_join(
+/// This function is public so other downstream projects can use it
+/// to construct `HashJoinExec` with right side as the build side.
```

Comment:
```suggestion
```

Reply:
I think there is a benefit in leaving the documentation that explains the rationale for why this is public